executor/parallel_apply_test.go

```diff
@@ -0,0 +1,112 @@
+|           └─TopN_35                  | 1.00     | cop[tikv] |               | test.t.b:desc, offset:0, count:1                                                   |
+|             └─Selection_34           | 7992.00  | cop[tikv] |               | gt(test.t.a, test.t.a), not(isnull(test.t.b))                                      |
+|               └─TableFullScan_33     | 10000.00 | cop[tikv] | table:t2      | keep order:false, stats:pseudo                                                     |`
+	checkApplyPlan(c, tk, q1, p1Seria)
+	tk.MustQuery(q1).Sort().Check(testkit.Rows("1", "2", "3", "4", "5", "6", "7", "8", "9"))
+	tk.MustExec("set tidb_enable_parallel_apply=true")
+	p1Paral := `| Projection_15                        | 9990.00  | root      |               | test.t.b                                                                           |
+| └─Apply_17                           | 9990.00  | root      |               | CARTESIAN inner join, other cond:gt(test.t.b, Column#7) |
+|   ├─TableReader_20(Build)            | 9990.00  | root      |               | data:Selection_19                                                                  |
+|   │ └─Selection_19                   | 9990.00  | cop[tikv] |               | not(isnull(test.t.b))                                                              |
+|   │   └─TableFullScan_18             | 10000.00 | cop[tikv] | table:t1      | keep order:false, stats:pseudo                                                     |
+|   └─Selection_21(Probe)              | 0.80     | root      |               | not(isnull(Column#7))                                                              |
+|     └─StreamAgg_26                   | 1.00     | root      |               | funcs:max(test.t.b)->Column#7                                                      |
+|       └─TopN_27                      | 1.00     | root      |               | test.t.b:desc, offset:0, count:1                                                   |
+|         └─TableReader_36             | 1.00     | root      |               | data:TopN_35                                                                       |
+|           └─TopN_35                  | 1.00     | cop[tikv] |               | test.t.b:desc, offset:0, count:1                                                   |
+|             └─Selection_34           | 7992.00  | cop[tikv] |               | gt(test.t.a, test.t.a), not(isnull(test.t.b))                                      |
+|               └─TableFullScan_33     | 10000.00 | cop[tikv] | table:t2      | keep order:false, stats:pseudo                                                     |`
+	checkApplyPlan(c, tk, q1, p1Paral)
+	tk.MustQuery(q1).Sort().Check(testkit.Rows("1", "2", "3", "4", "5", "6", "7", "8", "9"))
+
+	// Apply_45 is in the inner side of Apply_28
```

Comment:
Can we have another way not to directly compare the explain result?

Reply:
OK, I update the check method `checkApplyPlan`.
The implementation of it is too simple to cover some complex cases like one `Apply` being the outer side of another `Apply`.
Since this PR is too big, I prefer to keep the test simple in this PR, and in the next PR, I will improve `checkApplyPlan` and add more tests about parallel `Apply`, and finally enable this feature by default.